### DISCUSSION TITLE
Checking for and calling another extension

### DISCUSSION THREAD

**Discussion by cmBarsotti** [microsoft/vscode-discussions#61] [Extension Development QnA] [answered]
Hello,
I have a sidebar VS Code extension I've written for my company. I'm trying to figure out if I can make sure the user had a specific extension installed and, if so, can I call that extensions commands from my extension.

The extension in question is MJML (published by MJML) mjmlio.vscode-mjml, and I would like to be able to call 'MJML: Open Preview to the Side' and others.

Thank you in advance.

**Comment by bwateratmsft** [comment, marked as answer]
Yes! The APIs you're interested in are [here](https://code.visualstudio.com/api/references/vscode-api#extensions). Whatever an extension returns from its `activate()` method, will be visible to other extensions as the `.exports` field on the result from `getExtension`.

**Comment by r3m0t** [comment]
The command IDs are listed in the extension gallery, under "feature contributions". You can run a command using, eg,  `await vscode.commands.executeCommand("mjml.foo")`.
